core/network/src/protocol.rs

```diff
@@ -852,13 +852,22 @@ impl<B: BlockT, S: NetworkSpecialization<B>, H: ExHashT> Protocol<B, S, H> {
 				network_out.disconnect_peer(who);
 				return;
 			}
+
 			if self.config.roles.is_light() {
+				// we're not interested in light peers
+				if status.roles.is_light() {
+					debug!(target: "sync", "Peer {} is unable to serve light requests", who);
+					network_out.report_peer(who.clone(), i32::min_value());
+					network_out.disconnect_peer(who);
+					return;
+				}
+
+				// we don't interested in peers that are far behind us
 				let self_best_block = self
 					.context_data
 					.chain
 					.info()
-					.best_queued_number
-					.unwrap_or_else(|| Zero::zero());
+					.chain.best_number;
```

Comment:
Just in case anyone has an opinion on this: `Client` always returns `None` in `info().best_queued_number`. Could we probably remove this field at all or fill it with `best_number`? Queueing occurs at the different level => `Client` knows nothing about it.

Reply:
I'm fine with removing it

Reply:
Removed `best_queued_*`. We could probably remove `ClientInfo` structure at all (since now if has only `chain` information), but that could be made in follow-up PRs. Don't want to alter a ton of code here